lava/magma/core/model/py/ports.py

```diff
@@ -55,16 +59,25 @@ def peek(self) -> np.ndarray:
 class PyInPortVectorSparse(PyInPort):
     """Python implementation of Vector Sparse InPort
     """
+    # sparse port recieves twice, for val and index.
```

Comment:
I don't think it needs to receive twice. The sender should probably interleave index and value vectors as described in pptx. This way both vector-valued and scalar-valued recv() functions can deal with the data.

Probably best to write unit tests first to convince yourself of the usage model.